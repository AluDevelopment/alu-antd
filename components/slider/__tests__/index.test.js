import React from 'react';
import { render, mount } from 'enzyme';
import Slider from '..';

describe('Slider', () => {
  it('should show tooltip when hovering slider handler', () => {
    const wrapper = mount(<Slider defaultValue={30} />);
    wrapper
      .find('.alu-slider-handle')
      .at(0)
      .simulate('mouseEnter');
    expect(
      render(
        wrapper
          .find('Trigger')
          .instance()
          .getComponent(),
      ),
    ).toMatchSnapshot();
    wrapper
      .find('.alu-slider-handle')
      .at(0)
      .simulate('mouseLeave');
    expect(
      render(
        wrapper
          .find('Trigger')
          .instance()
          .getComponent(),
      ),
    ).toMatchSnapshot();
  });

  it('when tooltipVisible is true, tooltip should show always, or should never show', () => {
    let wrapper = mount(<Slider defaultValue={30} tooltipVisible />);
    expect(
      wrapper
        .find('.alu-tooltip-content')
        .at(0)
        .hasClass('alu-tooltip-hidden'),
    ).toBe(false);
    wrapper
      .find('.alu-slider-handle')
      .at(0)
      .simulate('mouseEnter');
    expect(
      wrapper
        .find('.alu-tooltip-content')
        .at(0)
        .hasClass('alu-tooltip-hidden'),
    ).toBe(false);
    wrapper
      .find('.alu-slider-handle')
      .at(0)
      .simulate('click');
    expect(
      wrapper
        .find('.alu-tooltip-content')
        .at(0)
        .hasClass('alu-tooltip-hidden'),
    ).toBe(false);
    wrapper = mount(<Slider defaultValue={30} tooltipVisible={false} />);
    expect(wrapper.find('.alu-tooltip-content').length).toBe(0);
  });
});
