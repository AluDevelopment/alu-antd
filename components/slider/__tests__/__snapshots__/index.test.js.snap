// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should show tooltip when hovering slider handler 1`] = `
<div>
  <div
    class="alu-tooltip  alu-tooltip-placement-top "
  >
    <div
      class="alu-tooltip-content"
    >
      <div
        class="alu-tooltip-arrow"
      />
      <div
        class="alu-tooltip-inner"
        role="tooltip"
      >
        30
      </div>
    </div>
  </div>
</div>
`;

exports[`Slider should show tooltip when hovering slider handler 2`] = `
<div>
  <div
    class="alu-tooltip  alu-tooltip-placement-top  alu-tooltip-hidden"
  >
    <div
      class="alu-tooltip-content"
    >
      <div
        class="alu-tooltip-arrow"
      />
      <div
        class="alu-tooltip-inner"
        role="tooltip"
      >
        30
      </div>
    </div>
  </div>
</div>
`;
